patches/minecraft/net/minecraft/entity/Entity.java.patch

```diff
@@ -356,6 +356,57 @@
 +            && (this instanceof EntityPlayer || world.func_82736_K().func_82766_b("mobGriefing"))
 +            && this.field_70130_N * this.field_70130_N * this.field_70131_O > 0.512F;
 +    }
++
++    /**
++     * A functional call to consume a shared list of item drops, usually
++     * to collect all drops, modify, and or clear, before throwing an event.
++     * This is to be used instead of manually flipping fields. The provided
++     * list is always cleared on each call. If an event is to be thrown, the
++     * event should be thrown in the provided {@link java.util.function.Consumer}
++     * @param droppedItems
++     */
++    @SuppressWarnings("deprecation")
++    public void captureDrops(java.util.function.Consumer<List<EntityItem>> droppedItems)
++    {
++        if (this.performingDropCaptures) {
++            // verify someone isn't already using the field
++            // If they are, pass the same field over as technically
++            // the list will clear after the previous usage is completed.
++            droppedItems.accept(this.capturedDrops);
++            return;
++        }
++        this.performingDropCaptures = true; // actual used field
++        this.capturedDrops.clear(); // always clear for preparation,
++        // consumer code should always use the provided list
++        droppedItems.accept(this.capturedDrops);
++        // clear again, just in case. Can't
++        this.capturedDrops.clear();
++        this.performingDropCaptures = false; // actual used field
++    }
++
++    public boolean isCapturingItemDrops()
++    {
++        return this.performingDropCaptures;
++    }
++
++    /**
++     * Uses the provided supplier to artificially capture and remove the capture.
++     * Use this method in the case where something can provide an {@link EntityItem},
++     * such as {@link net.minecraft.entity.player.EntityPlayer#dropItem(Item, int)}
++     * where the item dropped needs to be provided, but not actually spawned, and
++     * therefor "captured"; however, the intention is to simply provide the entity
++     * without capturing the entity, for later spawning.
++     *
++     * @param supplier The supplier of the entity item
++     * @return The entity item from the supplier, without spawning or capturing
++     */
++    public EntityItem captureDrop(java.util.function.Supplier<EntityItem> supplier)
```

Comment:
Since the use is only for the ForgeHooks side, I intended this to simply be a "ignore capture and spawn" method, to avoid exposing the underlying list and boolean flags, but some discussion could be used.

Reply:
Why do we use a supplier? The getting is not delayed or conditioned, so I think you can just pass an entity instance in.